x-pack/plugins/watcher/public/sections/watch_list/components/watch_list.tsx

```diff
@@ -0,0 +1,267 @@
+          </Fragment>
+        );
+      },
+    },
+    {
+      field: 'watchStatus.comment',
+      name: i18n.translate('xpack.watcher.sections.watchList.watchTable.commentHeader', {
+        defaultMessage: 'Comment',
+      }),
+      sortable: true,
+      truncateText: true,
+    },
+    {
+      field: 'watchStatus.lastMetCondition',
+      name: i18n.translate('xpack.watcher.sections.watchList.watchTable.lastFiredHeader', {
+        defaultMessage: 'Last fired',
+      }),
+      sortable: true,
+      truncateText: true,
+      render: (lastMetCondition: Moment) => {
+        return lastMetCondition ? lastMetCondition.fromNow() : lastMetCondition;
+      },
+    },
+    {
+      field: 'watchStatus.lastChecked',
+      name: i18n.translate('xpack.watcher.sections.watchList.watchTable.lastTriggeredHeader', {
+        defaultMessage: 'Last triggered',
+      }),
+      sortable: true,
+      truncateText: true,
+      render: (lastChecked: Moment) => {
+        return lastChecked ? lastChecked.fromNow() : lastChecked;
+      },
+    },
+    {
+      actions: [
+        {
+          render: watch => {
+            const disabled = watch.isSystemWatch || watch.isBeingDeleted;
+            return (
+              <EuiButtonEmpty
+                iconType="pencil"
+                disabled={disabled}
+                aria-label={intl.formatMessage({
+                  id: 'xpack.watcher.sections.watchList.watchTable.menuEditButtonDescription',
+                  defaultMessage: 'Edit watch',
+                })}
+                onClick={() => {
+                  urlService.change(
+                    `/management/elasticsearch/watcher/watches/watch/${watch.id}/edit`
+                  );
+                }}
+              >
+                <FormattedMessage
+                  id="xpack.watcher.sections.watchList.watchTable.menuEditButtonTitle"
+                  defaultMessage="Edit"
+                />
+              </EuiButtonEmpty>
+            );
+          },
+        },
+      ],
+    },
+  ];
+  const selectionConfig = {
+    onSelectionChange: setSelection,
+  };
+  const pagination = {
+    initialPageSize: 10,
+    pageSizeOptions: [10, 50, 100],
+  };
+  const searchConfig = {
+    box: {
+      incremental: true,
+    },
+    toolsRight: (
+      <EuiButton
+        data-test-subj="btnDeleteWatches"
+        onClick={() => {
+          setWatchesToDelete(selection.map(selected => selected.id));
+        }}
+        color="danger"
+        disabled={!selection.length}
+      >
+        <FormattedMessage
+          id="xpack.watcher.sections.watchList.deleteWatchButtonLabel"
+          defaultMessage="Delete"
+        />
+      </EuiButton>
+    ),
+  };
+  return (
+    <EuiPageContent>
+      <DeleteWatchesModal
+        callback={(deleted?: string[]) => {
+          if (deleted) {
+            setWatches(
+              watches.filter(watch => {
+                return !deleted.includes(watch.id);
```

Comment:
I'm seeing this throw an error when I hit "Cancel". I believe this is due to it receiving a `SyntheticEvent`. In the modal we probably just need to call `callback()` directly:

```js
onCancel={() => callback()}
```

![image](https://user-images.githubusercontent.com/1238659/53759571-8087f400-3e75-11e9-98ee-46a36b389454.png)

Reply:
fixed